### DISCUSSION TITLE
## 1.13.0 / 2022-08-06

### DISCUSSION THREAD

**Discussion by bwplotka** [prometheus/client_golang#1112] [Announcements]
* [CHANGE] Minimum required Go version is now 1.17 (we also test client_golang against the new 1.19 version).
* [ENHANCEMENT] Added `prometheus.TransactionalGatherer` interface for `promhttp.Handler` use which allows using low allocation update techniques for custom collectors. #989
* [ENHANCEMENT] Added exemplar support to `prometheus.NewConstHistogram`. See [`ExampleNewConstHistogram_WithExemplar`](prometheus/examples_test.go#L602) example on how to use it. #986
* [ENHANCEMENT] `prometheus/push.Pusher` now has context-aware methods that pass context to HTTP requests. #1028
* [ENHANCEMENT] `prometheus/push.Pusher` has now `Error` method that retrieve last error. #1075
* [ENHANCEMENT] `testutil.GatherAndCompare` provides now readable diff on failed comparisons. #998
* [ENHANCEMENT] Query API now supports timeouts. #1014
* [ENHANCEMENT] New `MetricVec` method `DeletePartialMatch(labels Labels)` for deleting all metrics that match provided labels. #1013
* [ENHANCEMENT] `api.Config` now accepts passing custom `*http.Client`. #1025
* [BUGFIX] Raise exemplar labels limit from 64 to 128 bytes as specified in OpenMetrics spec. #1091
* [BUGFIX] Allow adding exemplar to +Inf bucket to const histograms. #1094
* [ENHANCEMENT] Most `promhttp.Instrument*` middlewares now support adding exemplars to metrics. This allows hooking those to your tracing middleware that retrieves trace ID and puts it in exemplar if present. #1055
* [ENHANCEMENT] Added `testutil.ScrapeAndCompare` method. #1043
* [BUGFIX] Fixed `GopherJS` build support. #897
* [ENHANCEMENT] :warning: Added way to specify what `runtime/metrics`  `collectors.NewGoCollector` should use. See [`ExampleGoCollector_WithAdvancedGoMetrics`](prometheus/collectors/go_collector_latest_test.go#L263). #1102

## New Contributors ❤️ 

* @alissa-tung made their first contribution in https://github.com/prometheus/client_golang/pull/1000
* @wperron made their first contribution in https://github.com/prometheus/client_golang/pull/986
* @zhijian-pro made their first contribution in https://github.com/prometheus/client_golang/pull/1005
* @dohnto made their first contribution in https://github.com/prometheus/client_golang/pull/1021
* @tatsuhiro-t made their first contribution in https://github.com/prometheus/client_golang/pull/1028
* @sourikghosh made their first contribution in https://github.com/prometheus/client_golang/pull/998
* @josephwoodward made their first contribution in https://github.com/prometheus/client_golang/pull/1014
* @stone-z made their first contribution in https://github.com/prometheus/client_golang/pull/1013
* @yolossn made their first contribution in https://github.com/prometheus/client_golang/pull/1025
* @xrstf made their first contribution in https://github.com/prometheus/client_golang/pull/1039
* @dependabot made their first contribution in https://github.com/prometheus/client_golang/pull/1051
* @sazary made their first contribution in https://github.com/prometheus/client_golang/pull/1056
* @oiooj made their first contribution in https://github.com/prometheus/client_golang/pull/1075
* @fredr made their first contribution in https://github.com/prometheus/client_golang/pull/1095
* @inosato made their first contribution in https://github.com/prometheus/client_golang/pull/1096
* @arun-shopify made their first contribution in https://github.com/prometheus/client_golang/pull/1094
* @paralin made their first contribution in https://github.com/prometheus/client_golang/pull/897

**Full Changelog**: https://github.com/prometheus/client_golang/compare/v1.12.1...v1.13.0

<hr /><em>This discussion was created from the release <a href='https://github.com/prometheus/client_golang/releases/tag/v1.13.0'>## 1.13.0 / 2022-08-06</a>.</em>

**Comment by hairyhenderson** [comment]
I'm unable to use v1.13.0 due to #1117 - can we get a v1.13.1 released soon with the fix (#1118) which was merged over a month ago?
